stock_release_channel_shipment_advice/models/stock_release_channel.py

```diff
@@ -0,0 +1,99 @@
+        readonly=True,
+    )
+
+    def button_show_shipment_advice(self):
+        self.ensure_one()
+        return {
+            "type": "ir.actions.act_window",
+            "name": _("Shipment Advice"),
+            "view_mode": "calendar,tree,form",
+            "res_model": self.shipment_advice_ids._name,
+            "domain": [("id", "in", self.shipment_advice_ids.ids)],
+            "context": self.env.context,
+        }
+
+    @api.depends("shipment_planning_method", "picking_to_plan_ids")
+    def _compute_can_plan_shipment(self):
+        for rec in self:
+            rec.can_plan_shipment = (
+                rec.picking_to_plan_ids
+                and rec.shipment_planning_method
+                and rec.shipment_planning_method != "none"
+            )
+
+    @api.depends("picking_ids", "picking_ids.planned_shipment_advice_id")
+    def _compute_picking_to_plan_ids(self):
+        for rec in self:
+            rec.picking_to_plan_ids = rec.picking_ids.filtered(
+                lambda p: not p.planned_shipment_advice_id
+            )
+
+    def button_plan_shipment(self):
+        self.ensure_one()
+        if not self.can_plan_shipment:
+            raise UserError(_("Shipment planning not allowed"))
+        shipment_advice = self._plan_shipment()
+        if not shipment_advice:
+            return {}
+        return {
+            "type": "ir.actions.act_window",
+            "name": _("Shipment Advice"),
+            "view_mode": "form",
+            "res_model": shipment_advice._name,
+            "res_id": shipment_advice.id,
+            "context": self.env.context,
+        }
+
+    def _plan_shipment(self):
+        self.ensure_one()
+        if not self.shipment_planning_method or self.shipment_planning_method == "none":
+            return self.env["shipment.advice"]
+        _shipment_planning_method_name = self._get_shipment_planning_method_name()
+        if not hasattr(self, _shipment_planning_method_name):
+            raise UserError(
+                _("There is no implementation for the shipment planning method '%s'")
+                % self.shipment_planning_method
+            )
+        return getattr(self, _shipment_planning_method_name)()
+
+    def _plan_shipment_simple(self):
+        """Meant to be inherited in case we would like to redirect."""
+        self.ensure_one()
+        shipment_advice_model = self.env["shipment.advice"]
+        if not self.picking_to_plan_ids:
+            return shipment_advice_model
+        return shipment_advice_model.create(self._prepare_shipment_advice_vals())
+
+    def _prepare_shipment_advice_vals(self):
+        self.ensure_one()
+        return {
+            "release_channel_id": self.id,
+            "planned_move_ids": [Command.set(self.picking_to_plan_ids.move_ids.ids)],
+            "shipment_type": "outgoing",
+            "warehouse_id": self.warehouse_id.id,
+            "company_id": self.warehouse_id.company_id.id,
+        }
```

Comment:
This method should be split in 2 methods to ease extension:
* `_prepare_shipment_advice_common_vals` 
* `_prepare_shipment_advice_simple_vals` 

```python

    def _prepare_shipment_advice_common_vals(self):
        self.ensure_one()
        return {
            "release_channel_id": self.id,
            "shipment_type": "outgoing",
            "warehouse_id": self.warehouse_id.id,
            "company_id": self.warehouse_id.company_id.id,
        }

    def _prepare_shipment_advice_simple_vals(self):
        vals = self._prepare_shipment_advice_common_vals()
        vals["planned_move_ids"] = [Command.set(self.picking_to_plan_ids.move_ids.ids)]
        return vals
```